File: java/src/dsclient/scheduling_policies/AllToLargest.java
package dsclient.scheduling_policies;

import java.io.IOException;
import java.util.Collections;
import java.util.Comparator;

import dsclient.JobDispatchPolicy;
import dsclient.JobScheduler;
import dsclient.models.JobSubmission;

public class AllToLargest implements JobDispatchPolicy {
    public void dispatch(JobScheduler scheduler) throws IOException {
        String largestServerType = Collections.max(
            scheduler.servers,
            Comparator.comparing(srv -> srv.coreCount)
        ).type;

        while (true) {
            String received = scheduler.inquire("REDY");
            if (received.equals("NONE")) {
                break;
            }

            JobSubmission job = JobSubmission.fromReceivedLine(received);

            scheduler.inquire(String.format("SCHD %d %s 0", job.id, largestServerType));
        }
    }
}

File: python/ds_client/scheduling_policies/all_to_largest.py

from __future__ import annotations
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from ..job_scheduler import JobScheduler

from ..job_dispatch_policy import JobDispatchPolicy
from ..models import JobSubmission

class AllToLargest(JobDispatchPolicy):
    def dispatch(self, scheduler: JobScheduler) -> None:
        largest_server_type = max(
            scheduler.servers,
            key=lambda srv: srv.core_count,
        ).type

        while True:
            received = scheduler.inquire('REDY')
            if received == 'NONE':
                break

            job = JobSubmission.from_received_line(received)

            scheduler.inquire(f"SCHD {job.id} {largest_server_type} 0")
